src/steiner_tree.rs

```diff
@@ -121,6 +123,110 @@ fn _metric_closure_edges(
     Ok(out_vec)
 }
 
+fn fast_metric_edges(
```

Comment:
Besides getting different results for the user api is there a reason we don't want to just replace `_metric_closure_edges` with this implementation? If so we should probably put that in a comment here.

Reply:
Oh, no, these functions compute different things. 
`_metric_closure_edges`  computes the shortest path between all pairs of nodes while `fast_metric_edges` computes the shortest path between all pairs `(s, t)` of the given `terminal_nodes` *provided* that:
- there is an edge `(u, v)` in the graph.
- node `s` is the closest node to  `u` among all `terminal_nodes`
- node `t`  is the closest node to `v` among all `terminal_nodes`

For example, if  all vertices are terminals, `fast_metric_edges` will return the original edges of the graph.

Probably, the name of this function was misleading. I'll add this explanation in a comment.